include/cyclic_reduction.hxx

```diff
@@ -88,43 +91,44 @@ public:
   /// By default not periodic
   void setPeriodic(bool p=true) {periodic=p;}
   
-  /// Set up a single equation to invert
-  void setCoefs(T a[], T b[], T c[]) {
-    setCoefs(1, &a, &b, &c);
-  }
-
   /// Set the entries in the matrix to be inverted
   ///
   /// @param[in] nsys   The number of independent matrices to be solved
   /// @param[in] a   Left diagonal. Should have size [nsys][N]
   ///                where N is set in the constructor or setup
   /// @param[in] b   Diagonal values. Should have size [nsys][N]
   /// @param[in] c   Right diagonal. Should have size [nsys][N]
-  void setCoefs(int nsys, T **a, T **b, T **c) {
+  void setCoefs(Matrix<T> &a, Matrix<T> &b, Matrix<T> &c) {
+    TRACE("CyclicReduce::setCoefs");
+    
+    int nsys = std::get<0>(a.shape());
+    
     // Make sure correct memory arrays allocated
     allocMemory(nprocs, nsys, N);
 
     // Fill coefficient array
     for(int j=0;j<Nsys;j++)
       for(int i=0;i<N;i++) {
-        coefs[j][4*i] = a[j][i];
-        coefs[j][4*i + 1] = b[j][i];
-        coefs[j][4*i + 2] = c[j][i];
+        coefs(j, 4*i) = a(j, i);
+        coefs(j, 4*i + 1) = b(j, i);
+        coefs(j, 4*i + 2) = c(j, i);
         // 4*i + 3 will contain RHS
       }
   }
 
-  /// Solve a single triadiagonal system
-  /// 
-  void solve(T rhs[], T x[]) {
-    // Solving single system
-    solve(1, &rhs, &x);
-  }
+  // /// Solve a single triadiagonal system
+  // /// 
+  // void solve(Matrix<T> &rhs, Matrix<T> &x) {
+  //   // Solving single system
+  //   solve(1, &rhs, &x);
+  // }
```

Comment:
Commented-out code should be removed instead

Reply:
I thought I'd not staged this, oops.